src/main/java/shukaro/warptheory/handlers/IWarpEvent.java

```diff
@@ -13,13 +13,12 @@
 
     public final int getCost() { return (int)Math.ceil(getSeverity() / (double)10); }
 
-	public boolean canDo(EntityPlayer player) {
-		boolean flag = true;
-		flag = !player.isPotionActive(potionWarpWardID) || (WarpHandler.getUnavoidableCount(player) > 0);
-		WarpHandler.addUnavoidableCount(player, -1);
-		flag &= !(wuss||player.capabilities.isCreativeMode);
-		return flag;
-	}
+    public boolean canDo(EntityPlayer player)
+    {
+        boolean flag = !player.isPotionActive(potionWarpWardID) || (WarpHandler.getUnavoidableCount(player) > 0) && !(wuss||player.capabilities.isCreativeMode);
+        WarpHandler.addUnavoidableCount(player, -1);
```

Comment:
I suddenly get skeptic about this line. Shouldn't it be in `doEvent` or in `WarpEventHandler` instead?

Reply:
Well, I just made the code according to the original author's specifications. `doEvent` only handles the actual effect, and it's abstract method. `canDo` is the function which actually check if the event effect can be applied to the player, so I think this is much suitable than `WarpEventHandler`.

Reply:
> I suddenly get skeptic about this line. Shouldn't it be in `doEvent` or in `WarpEventHandler` instead?

Well, I just made the code according to the original author's specifications. `doEvent` only handles the actual effect, and it's abstract method. `canDo` is the function which actually check if the event effect can be applied to the player, so I think this is much suitable than `WarpEventHandler`.

Reply:
Anyhow, the behavior should be correct. It just bugs me a bit that stuff got mutated in something that sounds like a immutable query.